### ISSUE TITLE
RabbitMQ GPG error during provisioning

### ISSUE THREAD

**Issue by hackdna** [parklab/refinery-platform#1181]
Commit: 30a5c4239e304383c45a7234f66c5748758cc9e8

### Steps to reproduce
If you have an existing Vagrant VM, run `vagrant provision`. Otherwise:
```shell
$ git clone git@github.com:parklab/refinery-platform.git
$ cd refinery-platform
$ vagrant up
```
or spin up an AWS stack.

### Observed behavior
In the first case, the following warning is displayed in the console:
```
==> default: stdin: is not a tty
==> default: W
==> default: :
==> default: An error occurred during the signature verification. The repository is not updated and the previous index files will be used. GPG error: http://www.rabbitmq.com testing InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 6B73A36E6026DFCA
==> default: W
==> default: :
==> default: Failed to fetch http://www.rabbitmq.com/debian/dists/testing/InRelease
==> default: W
==> default: :
==> default: Some index files failed to download. They have been ignored, or old ones used instead.
```
In the second case, VM fails to get fully provisioned with the following errors displayed in the console:
```
==> default: Error: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install rabbitmq-server' returned 100: Reading package lists...
==> default: Building dependency tree...
==> default: Reading state information...
==> default: The following packages were automatically installed and are no longer required:
==> default:   libxfont1 libxkbfile1 x11-xkb-utils xfonts-base xfonts-encodings
==> default:   xfonts-utils xserver-common xserver-xorg-core
==> default: Use 'apt-get autoremove' to remove them.
==> default: The following extra packages will be installed:
==> default:   erlang-asn1 erlang-base erlang-corba erlang-crypto erlang-diameter
==> default:   erlang-edoc erlang-eldap erlang-erl-docgen erlang-eunit erlang-ic
==> default:   erlang-inets erlang-mnesia erlang-nox erlang-odbc erlang-os-mon
==> default:   erlang-parsetools erlang-percept erlang-public-key erlang-runtime-tools
==> default:   erlang-snmp erlang-ssh erlang-ssl erlang-syntax-tools erlang-tools
==> default:   erlang-webtool erlang-xmerl libodbc1 socat
==> default: Suggested packages:
==> default:   erlang erlang-manpages erlang-doc xsltproc fop erlang-ic-java
==> default:   erlang-observer libmyodbc odbc-postgresql tdsodbc unixodbc-bin
==> default: The following NEW packages will be installed:
==> default:   erlang-asn1 erlang-base erlang-corba erlang-crypto erlang-diameter
==> default:   erlang-edoc erlang-eldap erlang-erl-docgen erlang-eunit erlang-ic
==> default:   erlang-inets erlang-mnesia erlang-nox erlang-odbc erlang-os-mon
==> default:   erlang-parsetools erlang-percept erlang-public-key erlang-runtime-tools
==> default:   erlang-snmp erlang-ssh erlang-ssl erlang-syntax-tools erlang-tools
==> default:   erlang-webtool erlang-xmerl libodbc1 rabbitmq-server socat
==> default: 0 upgraded, 29 newly installed, 0 to remove and 170 not upgraded.
==> default: Need to get 24.1 MB of archives.
==> default: After this operation, 41.8 MB of additional disk space will be used.
==> default: WARNING: The following packages cannot be authenticated!
==> default:   rabbitmq-server
==> default: E: There are problems and -y was used without --force-yes
==> default:
==> default: Error: /Stage[main]/Rabbitmq::Install/Package[rabbitmq-server]/ensure: change from purged to present failed: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install rabbitmq-server' returned 100: Reading package lists...
==> default: Building dependency tree...
==> default: Reading state information...
==> default: The following packages were automatically installed and are no longer required:
==> default:   libxfont1 libxkbfile1 x11-xkb-utils xfonts-base xfonts-encodings
==> default:   xfonts-utils xserver-common xserver-xorg-core
==> default: Use 'apt-get autoremove' to remove them.
==> default: The following extra packages will be installed:
==> default:   erlang-asn1 erlang-base erlang-corba erlang-crypto erlang-diameter
==> default:   erlang-edoc erlang-eldap erlang-erl-docgen erlang-eunit erlang-ic
==> default:   erlang-inets erlang-mnesia erlang-nox erlang-odbc erlang-os-mon
==> default:   erlang-parsetools erlang-percept erlang-public-key erlang-runtime-tools
==> default:   erlang-snmp erlang-ssh erlang-ssl erlang-syntax-tools erlang-tools
==> default:   erlang-webtool erlang-xmerl libodbc1 socat
==> default: Suggested packages:
==> default:   erlang erlang-manpages erlang-doc xsltproc fop erlang-ic-java
==> default:   erlang-observer libmyodbc odbc-postgresql tdsodbc unixodbc-bin
==> default: The following NEW packages will be installed:
==> default:   erlang-asn1 erlang-base erlang-corba erlang-crypto erlang-diameter
==> default:   erlang-edoc erlang-eldap erlang-erl-docgen erlang-eunit erlang-ic
==> default:   erlang-inets erlang-mnesia erlang-nox erlang-odbc erlang-os-mon
==> default:   erlang-parsetools erlang-percept erlang-public-key erlang-runtime-tools
==> default:   erlang-snmp erlang-ssh erlang-ssl erlang-syntax-tools erlang-tools
==> default:   erlang-webtool erlang-xmerl libodbc1 rabbitmq-server socat
==> default: 0 upgraded, 29 newly installed, 0 to remove and 170 not upgraded.
==> default: Need to get 24.1 MB of archives.
==> default: After this operation, 41.8 MB of additional disk space will be used.
==> default: WARNING: The following packages cannot be authenticated!
==> default:   rabbitmq-server
==> default: E: There are problems and -y was used without --force-yes
==> default:
==> default: Notice: /Stage[main]/Rabbitmq::Config/File[/etc/rabbitmq]: Dependency Package[rabbitmq-server] has failures: true
==> default: Warning: /Stage[main]/Rabbitmq::Config/File[/etc/rabbitmq]: Skipping because of failed dependencies
==> default: Notice: /Stage[main]/Rabbitmq::Config/File[rabbitmq.config]: Dependency Package[rabbitmq-server] has failures: true
==> default: Warning: /Stage[main]/Rabbitmq::Config/File[rabbitmq.config]: Skipping because of failed dependencies
==> default: Notice: /Stage[main]/Rabbitmq::Config/File[/etc/default/rabbitmq-server]: Dependency Package[rabbitmq-server] has failures: true
==> default: Warning: /Stage[main]/Rabbitmq::Config/File[/etc/default/rabbitmq-server]: Skipping because of failed dependencies
==> default: Notice: /Stage[main]/Rabbitmq::Config/File[rabbitmqadmin.conf]: Dependency Package[rabbitmq-server] has failures: true
==> default: Warning: /Stage[main]/Rabbitmq::Config/File[rabbitmqadmin.conf]: Skipping because of failed dependencies
```

### Expected behavior
VM should start and get provisioned without errors.

**Comment by hackdna** [comment]
Updating puppetlabs-rabbitmq to the latest version (5.4.0) causes [dependency cycle due to backwards compatibility broken with puppetlabs-apt 1.x](https://tickets.puppetlabs.com/browse/MODULES-2995).

Updating puppetlabs-apt to 2.x is non-trivial: https://github.com/parklab/refinery-platform/issues/774

**Comment by hackdna** [comment]
This is related to the key transition: https://www.rabbitmq.com/key-transition-0xF7B8CEA6056E8E56.txt However, the error message references the new key, not the old one.

**Comment by hackdna** [comment]
Fixed in f03ac118

**Comment by hackdna** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Troubleshoot neo4j key fingerprint mismatch error](https://trello.com/c/P0wawViL/87-troubleshoot-neo4j-key-fingerprint-mismatch-error)
